modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/common/LogInitialization.scala

```diff
@@ -6,73 +6,77 @@ import java.util.logging.LogManager
 import scala.io.{Codec, Source}
 import scalaz._
 import Scalaz._
-import scalaz.effect.IO
+import scalaz.concurrent.Task
 
-trait LogInitialization {
-  sealed trait LogConfError
-  case class ConfigurationNotFound(s: String) extends LogConfError
-  case class LogDirNotFound(f: File) extends LogConfError
-  case object ConfigurationLoadingError extends LogConfError
+trait AppBaseDir {
+  /**
+    * Calculates the base dir of the application based on the location of "this" class jar file
+    * It will throw an exception if unable to find the base dir
+    */
+  def baseDir: Task[File] = Task.delay {
+    val clazz = this.getClass
+    val fileName = clazz.getResource(s"/${clazz.getName.replace(".", System.getProperty("file.separator"))}.class").getFile
 
-  type LogConf[A] = EitherT[IO, LogConfError, A]
```

Comment:
It's very nice that in FP you can deal with errors statically, but the thing here is, that since you already have to handle exceptions raised from `IO`, `Task`s in this case, you don't gain much by having 2 ways of dealing with errors. Dealing statically with errors in the `EitherT` gives you the false sense of security that you are already dealing with exceptions but it's not true, since the IO errors can be raised at any moment and you still have to deal with them.

But this is a controversial topic that can generate many heated debates. I'm of the camp that `EitherT Error IO a` is usually a smell of:
 
1. You are dealing with genuine static errors where you shouldn't. The code dealing with `Either Error a` needs to be factored out from the `IO` code.

2. You are shoehorning exceptions that are meant to be handled in the traditional out-of-band way with `try`/`catch` blocks, like most IO errors.

Maybe I'm guilty of extrapolating my knowledge of Tasks too literally from Haskell `IO`, so I'm still open to change my opinion on this matter if it turns out my assumptions are wrong. For example, is it possible to properly handle asynchronous exceptions (like an interrupt anywhere in the code) statically within the `Task`s API? Can `Task`s guarantee there won't be any raised exceptions *leaking* through?